src/Text/Pandoc/Readers/Man.hs

```diff
@@ -0,0 +1,284 @@
+    "BI"   -> toBoldItalic
+    "IB"   -> toBoldItalic
+    "I"    -> toItalic
+    "IR"   -> toItalic
+    "RI"   -> toItalic
+    "SH"   -> return $ Header 2 nullAttr [Str joinedArgs]
+    "sp"   -> return $ if inCodeBlock rst then Null else Plain [LineBreak]
+    _      -> unkownMacro macroName
+
+  where
+
+  macroTitle :: PandocMonad m => String -> ManParser m Block
+  macroTitle mantitle = do
+    modifyState (changeTitle mantitle)
+    if null mantitle
+      then return Null
+      else return $ Header 1 nullAttr [Str mantitle]
+    where
+    changeTitle title mst@ManState{ pState = pst} =
+      let meta = stateMeta pst
+          metaUp = Meta $ insert "title" (MetaString title) (unMeta meta)
+      in
+      mst { pState = pst {stateMeta = metaUp} }
+
+  macroCodeBlock :: PandocMonad m => Bool -> ManParser m ()
+  macroCodeBlock insideCB = modifyRoffState (\rst -> rst{inCodeBlock = insideCB}) >> return ()
+
+  macroBR :: String -> Bool -> Block
+  macroBR txt inCode | inCode    = Plain [Code nullAttr txt]
+                     | otherwise = fromMaybe (Plain [Strong [Str txt]]) (linkToMan txt)
+
+  linkToMan :: String -> Maybe Block
+  linkToMan txt = case runParser linkParser () "" txt of
+      Right lnk -> Just $ Plain [lnk]
+      Left _ -> Nothing
+    where
+    linkParser :: Parsec String () Inline
+    linkParser = do
+      mpage <- many1 alphaNum
+      space
+      char '('
+      mansect <- digit
+      char ')'
+      -- assuming man pages are generated from Linux-like repository
+      let manurl pagename section = "../"++section++"/"++pagename++"."++section
+      return $ Link nullAttr [Str txt] (manurl mpage [mansect], mpage)
+
+    
+  unkownMacro :: PandocMonad m => String -> ManParser m Block
+  unkownMacro mname = do
+    pos <- getPosition
+    logMessage $ SkippedContent ("Unknown macro: " ++ mname) pos
+    return Null
+   
+  parseArgs :: PandocMonad m => ManParser m [String]
+  parseArgs = do
+    eolOpt <- optionMaybe $ char '\n'
+    if isJust eolOpt
+      then return []
+      else do
+        many1 space
+        arg <- try quotedArg <|> plainArg
+        otherargs <- parseArgs
+        return $ arg : otherargs
+
+    where
+
+    plainArg :: PandocMonad m => ManParser m String
+    plainArg = many1 $ noneOf " \t\n"
+
+    quotedArg :: PandocMonad m => ManParser m String
+    quotedArg = do
+      char '"'
+      val <- many1 quotedChar
+      char '"'
+      return val
+
+    quotedChar :: PandocMonad m => ManParser m Char
+    quotedChar = noneOf "\"\n" <|> try (string "\"\"" >> return '"')
+
+roffInline :: RoffState -> String -> [Inline]
+roffInline rst str
+  | null str && (not $ inCodeBlock rst) = []
+  | inCodeBlock rst = [Code nullAttr str]
+  | otherwise       = case fontKind rst of
+    Regular    -> [Str str]
+    Italic     -> [Emph [Str str]]
+    Bold       -> [Strong [Str str]]
+    ItalicBold -> [Emph [Strong [Str str]]]
+
+parseLine :: PandocMonad m => ManParser m Block
```

Comment:
Instead of having the parsers return a Block, you should have them return Blocks (from Text.Pandoc.Builder).  All of the readers are structured this way now.  Use functions from Text.Pandoc.Builder to create the return values, and use monoidal mappend to concatenate them.  A 'Blocks' is essentially a wrapped sequence of Block elements (perhaps empty).  This gives you a lot more flexibility:  a parser can return no blocks (mempty) or several.